CheapShopWeb/Services/Filtering.cs

```diff
@@ -51,28 +54,101 @@ public static List<Product> Filter(List<Product> productList, string name, strin
             List<Product> newlist= new List<Product>();
-            newlist.Add(prod);
-            string[] searchString = prod.name.Split(' ');
-            if (!string.IsNullOrEmpty(prod.name))
+            foreach (var product in productList)
             {
-                foreach (var product in productList)
+                if (LevenshteinDistance.Compute(product.name, prod.name) <= 3*prod.name.Split(' ').Length) //or 4*//
                 {
-                    
+                    newlist.Add(product);
                 }
             }
 
+            newlist.Sort((x, y) => String.Compare(x.price, y.price, StringComparison.Ordinal));
+            newlist.Insert(0, prod);
             return newlist;
         }
-        
-        public static Func<string, float> SToFFunc = num =>  //string to float function for lambda expression // makes number string comparable
+        public static List<Product> GetSimilarProductsGroup(List<Product> productList, Product prod, string itemsGroup)
         {
-            float n = float.Parse(num) * 100;
-            return n;
-        };
+
+            var method = typeof(SmallerGroups).GetMethod(itemsGroup + "Group");
+            if (!(method == null))
+            {
+                var smallerGroupList = (List<string>)method.Invoke(new SmallerGroups(), null); 
+                productList = productList.FindAll(product =>
+                        smallerGroupList.Any(group => product.group.ToLower().Equals(group.ToLower())));
+            } 
+            List<Product> newlist = new List<Product>();
+            foreach (var product in productList)
+            {
+                if (LevenshteinDistance.Compute(product.name, prod.name) <= 3 * prod.name.Split(' ').Length) //or 4*//
+                {
+                    newlist.Add(product);
+                }
+            }
+            newlist.Sort((x, y) => String.Compare(x.price, y.price, StringComparison.Ordinal));
+            newlist.Insert(0, prod);
+            return newlist;
+        }
+
+        public static Func<string, float> SToFFunc =
+            num => //string to float function for lambda expression // makes number string comparable
+            {
+                float n = float.Parse(num) * 100;
+                return n;
+            };
+    }
+    static class LevenshteinDistance
+    {
+        /// <summary>
+        /// Compute the distance between two strings.
+        /// </summary>
+        public static int Compute(string s, string t)
+        {
+            int n = s.Length;
+            int m = t.Length;
+            int[,] d = new int[n + 1, m + 1];
+
+            // Step 1
+            if (n == 0)
+            {
+                return m;
+            }
+
+            if (m == 0)
+            {
+                return n;
+            }
+
+            // Step 2
+            for (int i = 0; i <= n; d[i, 0] = i++)
```

Comment:
Čia neįgyvendinta, ar `d[i, 0] = i++` ir yra reikalignas algoritmas?

Reply:
Kodo aprašyme : "Initialization of matrix with row size source1Length and columns size source2Length"